share/completions/knife.fish

```diff
@@ -0,0 +1,829 @@
+#
+# Define functions for knife with arguments (e.g. knife bootstrap [--server])
+#
+
+
+function __fish_knife_scan_eq0
+  set cmd (commandline -opc)
+  if [ (count $cmd) -eq 1 ]
+    return 0
+  end
+  return 1
+end
+
+function __fish_knife_scan_eq1
+  set cmd (commandline -opc)
+  if [ (count $cmd) -eq 2 ]
+    if [ $argv[1] = $cmd[2] ]
+      return 0
+    end
+  end
+  return 1
+end
+
+function __fish_knife_scan_eq2
+  set cmd (commandline -opc)
+  set cmd_main $argv[1]
+  set cmd_sub $argv[2]
+
+  if [ (count $cmd) -eq 3 ]
+    if [ $cmd_main = $cmd[2] ]; and [ $cmd_sub = $cmd[3] ]
+      return 0
+    end
+  end
+  return 1
+end
+
+function __fish_knife_scan_eq3
+  set cmd (commandline -opc)
+  if [ (count $cmd) -eq 4 ]
+    if [ $argv[1] = $cmd[2] ]; and [ $argv[2] = $cmd[3] ]; and [ $argv[3] = $cmd[4] ]
```

Comment:
This will not perform as intended. See #1428 and #150.

Try either:
```fish
if begin; [ $argv[1] = $cmd[2] ]; and [ $argv[2] = $cmd[3] ]; and [ $argv[3] = $cmd[4] ]; end
```
or
```fish
if [ $argv[1] = $cmd[2] -a $argv[2] = $cmd[3] -a $argv[3] = $cmd[4] ]
```
(this may need some testing as I'm not certain about the precedence order of operations)